src/main/java/com/snowflake/kafka/connector/internal/streaming/SnowflakeSinkServiceV2.java

```diff
@@ -491,14 +498,44 @@ private void closeStreamingClient() {
 
   private void createTableIfNotExists(final String tableName) {
     if (this.conn.tableExist(tableName)) {
-      if (this.conn.isTableCompatible(tableName)) {
-        LOGGER.info("Using existing table {}.", tableName);
+      if (!this.enableSchematization) {
+        if (this.conn.isTableCompatible(tableName)) {
+          LOGGER.info("Using existing table {}.", tableName);
+        } else {
+          throw SnowflakeErrors.ERROR_5003.getException("table name: " + tableName);
+        }
       } else {
-        throw SnowflakeErrors.ERROR_5003.getException("table name: " + tableName);
+        this.conn.appendMetaColIfNotExist(tableName);
       }
     } else {
       LOGGER.info("Creating new table {}.", tableName);
-      this.conn.createTable(tableName);
+      if (this.enableSchematization) {
+        if (connectorConfig
+            .get("value.converter")
+            .equals("io.confluent.connect.avro.AvroConverter")) {
```

Comment:
can you extract this out into a new method? Usually the idea is to keep only one job in a method. (Also called single responsibility principle, mostly applied to classes but can also be applied to this method) https://www.baeldung.com/java-single-responsibility-principle

From the readers perspective, this are just too many if else cases.. 

I would may be also keep this in a separate class may be called SchematizationUtils. 

In this class you can keep a connection object for creating classes
schema registry for fetching schema 
and all other future schema evolution utils can be kept in this class. 

Benefits:
1. SRP
2. Unit Testing 
3. Readability